pkg/apis/psmdb/v1/psmdb_defaults.go

```diff
@@ -602,8 +625,31 @@ func (rs *ReplsetSpec) SetDefaults(platform version.Platform, cr *PerconaServerM
 		return fmt.Errorf("replset %s VolumeSpec: %v", rs.Name, err)
 	}
 
-	if rs.Expose.Enabled && rs.Expose.ExposeType == "" {
-		rs.Expose.ExposeType = corev1.ServiceTypeClusterIP
+	if rs.Expose.Enabled {
+		if rs.Expose.ExposeType == "" {
+			rs.Expose.ExposeType = corev1.ServiceTypeClusterIP
+
+			if rs.Expose.DeprecatedExposeType != "" {
+				if cr.CompareVersion("1.18.0") >= 0 {
+					log.Info("Field `.expose.exposeType` was deprecated in 1.18.0. Consider using `.expose.type` instead", "cluster", cr.Name, "namespace", cr.Namespace, "replset", rs.Name)
+				}
+				rs.Expose.ExposeType = rs.Expose.DeprecatedExposeType
+			}
+		}
+
+		if len(rs.Expose.DeprecatedServiceLabels) > 0 {
+			if cr.CompareVersion("1.18.0") >= 0 {
+				log.Info("Field `.expose.serviceLabels` was deprecated in 1.18.0. Consider using `.expose.labels` instead", "cluster", cr.Name, "namespace", cr.Namespace, "replset", rs.Name)
```

Comment:
I like this comment, for the user it should be obvious that the field is deprecated, but I'm afraid we can clutter the logs with this info.

But this got me thinking, we should utilize CRD deprecation approach, so when the CR is applied the user will see that it is deprecated and what to use instead. 

See here: https://kubernetes.io/docs/tasks/extend-kubernetes/custom-resources/custom-resource-definition-versioning/#version-deprecation

Also, I quickly checked how to do it with kubebuilder and looks like there is marker comment for this one, like this:
```
// +kubebuilder:deprecatedversion:warning="This field is deprecated. Please use NewField."
```

If you have time, maybe you could try this one out and we can start to use it with these deprecated fields. Later we can do for all of our deprecated field in all of our operators.

Reply:
The deprecation method you mentioned will deprecate the whole version instead of one field.

This would work if we had different `apiVersion`s for each release like we used to. However, we've decided to use just one version `psmdb.percona.com/v1` in this pull request: https://github.com/percona/percona-server-mongodb-operator/pull/988. 

The kubebuilder directive you mentioned will deprecate our current `psmdb.percona.com/v1` version, similar to what we currently have for old versions here: https://github.com/percona/percona-server-mongodb-operator/blob/d178dd65c74763bf8e3b00a6a36775ba117ab73e/deploy/crd.yaml#L674-L676

It would be better to make use of the version deprecation, but we need to reconsider our versioning process.

Currently, to reduce the amount of logs, I have decided to print them when cluster has `initializing` state: https://github.com/percona/percona-server-mongodb-operator/pull/1641/commits/df7592c819e1a48fc3832f97407a323cc7a6f97f

Reply:
Ok, I agree, I will create a task for this to investigate it more and see can we use it and will it help us.